tripal_chado/src/Plugin/Field/FieldType/ChadoPropertyTypeDefault.php

```diff
@@ -232,4 +232,84 @@ public function isCompatible(TripalEntityType $entity_type) : bool {
     return $compatible;
   }
 
+  /**
+   * {@inheritDoc}
+   * @see \Drupal\tripal\TripalField\Interfaces\TripalFieldItemInterface::discover()
+   */
+  public static function discover(TripalEntityType $bundle, string $field_id, array $field_definitions) : array{
+
+    /** @var \Drupal\tripal_chado\Database\ChadoConnection $chado **/
+    $chado = \Drupal::service('tripal_chado.database');
+
+    // Initialize with an empty field list.
+    $field_list = [];
+
+    // Make sure the base table setting exists.
+    $base_table = $bundle->getThirdPartySetting('tripal', 'chado_base_table');
+    if (!$base_table) {
+      return $field_list;
+    }
+
+    // Make sure the prop table exists in Chado.
+    $prop_table = $base_table . 'prop';
+    if (!$chado->schema()->tableExists($prop_table)) {
+      return $field_list;
+    }
+
+    // Search for all unique types in the prop table.
+    $query = $chado->select('1:' . $prop_table, 'pt');
+    $query->leftJoin('1:cvterm', 'cvt', 'pt.type_id = cvt.cvterm_id');
+    $query->leftJoin('1:dbxref', 'dbx', 'dbx.dbxref_id = cvt.dbxref_id');
+    $query->leftJoin('1:db', 'db', 'db.db_id = dbx.db_id');
+    $query->leftJoin('1:cv', 'cv', 'cv.cv_id = cvt.cv_id');
+    $query->addField('cvt', 'cvterm_id');
+    $query->addField('cvt', 'name', 'cvterm_name');
+    $query->addField('cvt', 'definition');
+    $query->addField('dbx', 'accession');
+    $query->addField('db', 'name', 'db_name');
+    $query->addField('cv', 'name', 'cv_name');
+    $results = $query->distinct()->execute()->fetchAll();
+
+    // Create a field entry for each property type.
+    foreach ($results as $recprop) {
+      $field_list[] = [
+        'name' => self::generateFieldName($bundle, $recprop->cvterm_name),
+        'content_type' => $bundle->getID(),
+        'label' => ucwords($recprop->cvterm_name),
+        'type' => self::$id,
+        'description' => $recprop->definition ? 'A record property with the following definition: ' . $recprop->definition : '',
```

Comment:
I think that both Lacey and I want the term included. How about this instead?
```suggestion
      $description = 'A record property with the following term: "' 
          . $recprop->cv_name . ':' . $recprop->cvterm_name . '"';
      if ($recprop->definition) {
        $description .= ' and definition: "' . $recprop->definition .'"';
      }
      $field_list[] = [
        'name' => self::generateFieldName($bundle, $recprop->cvterm_name),
        'content_type' => $bundle->getID(),
        'label' => ucwords($recprop->cvterm_name),
        'type' => self::$id,
        'description' => $description,
```
This will display as
![2024-05-28_termanddef](https://github.com/tripal/tripal/assets/8419404/a0eb9bf8-0ffc-4162-b7d5-041b675fc343)

Reply:
I am now realizing that the term name is in here twice. So maybe this is not quite right. If you prefer ignoring this suggestion it is fine with me.

Reply:
Ok, lets ignore this for now then and we can think about this on a per field basis as we go through and work on these discover functions.